src/sync-external-store--stateful-store/store.ts

```diff
@@ -0,0 +1,44 @@
+import { SyncExternalStore } from "../sync-external-store/index.js";
+import {
+  CreateErrorSnapshot,
+  CreateLoadingSnapshot,
+  CreateSuccessSnapshot,
+  createErrorSnapshot,
+  createLoadingSnapshot,
+  createSuccessSnapshot,
+} from "./create-snapshot.js";
+import { StoreState } from "./store-item.js";
+
+type UpdateStoreParams<TStore extends StoreState, TParams = void, TResponse = void> = {
+  key: keyof TStore;
+  params?: TParams;
+  service: (params: TParams) => Promise<TResponse>;
```

Comment:
Nit, rename this to `action` as it's not inherently tied to a service call. Also to make this more flexible consider a return type of `(params: TParams) => Promise<TResponse> | TResponse;` so it can handle sync and async even though the sync scenario doesn't make a ton of sense